dbms/src/Interpreters/Cluster.cpp

```diff
@@ -467,6 +467,7 @@ std::unique_ptr<Cluster> Cluster::getClusterWithMultipleShards(const std::vector
 
 Cluster::Cluster(const Settings & settings, const Cluster & from) : shards_info{}, addresses_with_failover{}
```

Comment:
@kiransunkari can you please briefly explain how circular topology is being achieved in the updated logic?  
As per my understanding,  I can see that we are trying to get each replica node of the shard and then trying to create an entry in connection pool. But I am not able to understand line number 480-483 because of my C++ skills.